baybe/utils/validation.py

```diff
@@ -68,3 +73,100 @@ def validator(self: Any, attribute: Attribute, value: Any) -> None:
 
 non_inf_float = _make_restricted_float_validator(allow_nan=True, allow_inf=False)
 """Validator for non-infinite floats."""
+
+
+def validate_target_input(data: pd.DataFrame, targets: Sequence[Target]) -> None:
+    """Validate input dataframe columns corresponding to targets.
+
+    Args:
+        data: The input dataframe to be validated.
+        targets: The allowed targets.
+
+    Raises:
+        ValueError: If the input dataframe is empty.
+        ValueError: If any target data contain NaN.
+        TypeError: If any numerical target data contain non-numeric values.
+        ValueError: If any binary target data contain values not part of the targets'
+            allowed values.
+    """
+    from baybe.targets import BinaryTarget, NumericalTarget
+
+    if len(data) < 1:
+        raise ValueError("The provided input dataframe cannot be emtpy.")
+
+    for t in targets:
+        if data[t.name].isna().any():
+            raise ValueError(
+                f"The target '{t.name}' has missing values or NaNs in the provided "
+                f"dataframe. Missing target values are not supported."
+            )
+
+        if isinstance(t, NumericalTarget):
+            if data[t.name].dtype.kind not in "iufb":
+                raise TypeError(
+                    f"The numerical target '{t.name}' has non-numeric entries in the "
+                    f"provided dataframe. Non-numeric target values are not supported."
+                )
+        elif isinstance(t, BinaryTarget):
+            if not (
+                data[t.name].isin(allowed := [t.failure_value, t.success_value]).all()
+            ):
+                raise ValueError(
+                    f"The binary target '{t.name}' has nvalid entries in the provided "
+                    f"dataframe. Allowed values are: {allowed}."
+                )
```

Comment:
Fix typo and more informative error message
```suggestion
        elif isinstance(t, BinaryTarget):
            allowed = {t.failure_value, t.success_value}
            if invalid := set(data[t.name].unique()) - allowed:
                raise ValueError(
                    f"The binary target '{t.name}' has invalid entries {invalid} "
                    f"in the provided dataframe. Allowed values are: {allowed}."
                )
```